// @shoelace-style/shoelace/dist/chunks/chunk.IFSNRA43.js
import {
  component_styles_default
} from "./chunk.FRNPSZRF.js";
import {
  e,
  n
} from "./chunk.L2RLCVJQ.js";
import {
  p,
  r,
  s
} from "./chunk.C3RD2ROC.js";
import {
  __decorateClass
} from "./chunk.IHGPZX35.js";

// src/components/tab-panel/tab-panel.styles.ts
var tab_panel_styles_default = r`
  ${component_styles_default}

  :host {
    --padding: 0;

    display: block;
  }

  .tab-panel {
    border: solid 1px transparent;
    padding: var(--padding);
  }
`;

// src/components/tab-panel/tab-panel.ts
var id = 0;
var SlTabPanel = class extends s {
  constructor() {
    super(...arguments);
    this.componentId = `tab-panel-${++id}`;
    this.name = "";
    this.active = false;
  }
  connectedCallback() {
    super.connectedCallback();
    this.id = this.id || this.componentId;
  }
  render() {
    this.style.display = this.active ? "block" : "none";
    return p`
      <div part="base" class="tab-panel" role="tabpanel" aria-hidden=${this.active ? "false" : "true"}>
        <slot></slot>
      </div>
    `;
  }
};
SlTabPanel.styles = tab_panel_styles_default;
__decorateClass([
  e({ reflect: true })
], SlTabPanel.prototype, "name", 2);
__decorateClass([
  e({ type: Boolean, reflect: true })
], SlTabPanel.prototype, "active", 2);
SlTabPanel = __decorateClass([
  n("sl-tab-panel")
], SlTabPanel);
var tab_panel_default = SlTabPanel;

export {
  tab_panel_default
};
